doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,352 @@
+        +---+---+                                                      |                |
+        | Hash  |                                                      |                |
+        +---+---+                                                      |                |
+            |                                                          |                |
+            v                                                          |                |
+       +---------+---------+---------+                                 |                |
+       | Bucket1 | ...     | BucketN |                                 v                |
+       +=========+=========+=========+                          +---------------+       |
+       | Header  | ...     | ...     |                          | OID           |<---------------+
+       +---------+---------+---------+                          | Num Chunk     |       |        |
+    +--| Entry0  | ...     | ...     |                          | <Offset, len> |       |        |
+    |  +---------+---------+---------+                          | <Offset, len> |       |        |
+    |  | ...     | ...     | ...     |                          | ...           |       |        |
+    |  +---------+---------+---------+                          +---------------+       |        |
+    |  | EntryK  | ...     | ...     |                                                  |        |
+    |  +---------+---------+---------+                                                  |        |
+    |                                                                                   |        |
+    |                                                                                   |        |
+    |  +---------------+  +----------+  +----------+                      segment       v        |
+    +->| object_info_t |  | omap     |  | data     |               +------------+------------+   |
+       +===============+  +==========+  +==========+          +--->| Block0     | Block1     |   |
+       | CID, OID      |  | CID, OID |  | CID, OID |          |    +============+============+   |
+       | Type          |  | Type     |  | Type     |          |    | Header     |            |   |
+       | Offset        |  | Offset   |  | Offset   |          |    +------------+            +   |
+       | Segment       |  | Segment  |  | Segment  |----------+    | Small data | Data       |   |
+       +---------------+  +----------+  +----------+               +------------+            +   |
+                                                                +--| OID meta Entries|       |   |
+                                                                |  +------------+------------+   |
+                                                                |                                |
+                                                                +--------------------------------+
+
+
+WAL
+---
+ToDo: need discussion before detailed design
```

Comment:
I'd like some kind of information at least about
1. Is there a single WAL for the whole store?  Does each AG get a WAL?  Are cross-WAL transactions possible?
2. What information goes into the WAL? (metadata updates, free space updates, data)?
3. How does the space consumed by the WAL interact with the free space tracking metadata?
...